helm/abm/files/conf.yaml

```diff
@@ -1,14 +1,59 @@
+    {{- $operation := $asset.capability }}
     {{- if and (.args) (eq (len .args) 1) -}}
-    {{- with (index .args 0) }}
-    {{- toYaml . | nindent 4 }}
+       {{- with (index .args 0) }}
+       {{- $format := .format }}
+         {{- /* output vault properties */ -}}
+         {{- if .vault }}
+           {{- range $k, $v := .vault }}
+             {{- if eq $k $operation }}
+                {{- print "vault_credentials:" | nindent 4 }}
+                {{- if $v.address }}
+                   {{- print "address: " $v.address | nindent 6 }}
+                {{- end }}
+                {{- if $v.authPath }}
+                   {{- print "authPath: " $v.authPath | nindent 6 }}
+                {{- end }}
+                {{- if $v.role }}
+                   {{- print "role: " $v.role  | nindent 6 }}
+                {{- end }}
+                {{- if $v.secretPath }}
+                   {{- print "secretPath: " $v.secretPath | nindent 6 }}
+                {{- end }}
+             {{- end }}
+           {{- end }}
+         {{- end }}
+         {{- print "connection:" | nindent 4 }}
+         {{- /* output the connection details. If additional properties defined in values.yaml then merge them. */ -}}
+         {{- $connectionName := .connection.name }}
+         {{- $connectionDetails := get .connection .connection.name }}
+         {{- if hasKey $connectorsMap $connectionName }}
+            {{- $additionalPropertiesForConnector := get $connectorsMap $connectionName }}
+            {{- if hasKey $additionalPropertiesForConnector $operation }}
+              {{- $additionalPropertiesForConnectorOp := get $additionalPropertiesForConnector $operation }}
+              {{- /* add keys according to the processing of the metadata properties. */ -}}
+              {{- if include "airbyte-module.is-format-equal" (dict "connectionFormat" $format "additionalProperties" $additionalPropertiesForConnectorOp) }}
+                {{- if hasKey $additionalPropertiesForConnectorOp "emit_format" -}}
+                  {{- $additionalPropertiesForConnectorOp := set $additionalPropertiesForConnectorOp "format" $format }}
+                {{- end }}
+                {{- if hasKey $additionalPropertiesForConnectorOp "emit_asset_key" -}}
+                    {{- $assetKey := get $additionalPropertiesForConnectorOp "emit_asset_key" -}}
+                   {{- $additionalPropertiesForConnectorOp := set $additionalPropertiesForConnectorOp $assetKey $asset.assetID }}
+                {{- end }}
+                {{- include "airbyte-module.remove-metadata-keys" $additionalPropertiesForConnectorOp }}
+                {{- $connectionDetails := merge $connectionDetails $additionalPropertiesForConnectorOp }}
+              {{- end }}
+            {{- end }}
+         {{- end }}
+         {{- toYaml .connection | nindent 6 }}
+       {{- end }}
     {{- end }}
-    {{- end }}
-{{- end -}}
+{{- end }}
 {{- else -}}
 data: []
 {{- end -}}
```

Comment:
Is it better to do the parsing here in `conf.yaml` or to pass all the values to the module and do the parsing in the module's code (as it was before the changes)?

Reply:
I think @cdoron preferred that the module code will not be affected by the connector type this is why all the translation is done in the chart.

Reply:
Can you add an example file of how this conf.yaml file might look? (just a suggestion, if it needs much work you can ignore it).

Reply:
@Mohammad-nassar10  `sample/read_mysql.yaml` is an example of a generated conf file.  ``helm install --generate-name --dry-run -f helm/abm/values.sample.read.mysql.yaml helm/abm` can be used to generate this sample. Please let me know if you think I should add more info. Thanks